pkg/daemon/update.go

```diff
@@ -959,9 +959,16 @@ func (dn *Daemon) update(oldConfig, newConfig *mcfgv1.MachineConfig, skipCertifi
 		return err
 	}
 
-	// Update the MCN's NodeNodeDegraded condition with the update result
 	defer func() {
+		// Update the MCN's NodeNodeDegraded condition with the update result
 		dn.reportMachineNodeDegradeStatus(retErr, pool)
+
+		// Report irreconcilable differences, if any
+		if dn.node != nil && retErr == nil {
+			if err := dn.irreconcilableReporter.CheckReportIrreconcilableDifferences(newConfig, dn.node.Name); retErr != nil {
```

Comment:
I'm fine with it here, but curious why we're doing it in the defer'ed function instead of the main loop. In theory, if the update itself was otherwise unsuccessful, we probably don't need to update this until it is? In case the user fixes their upgrade and never goes through the version we're checking here (since the defer would always run)